### PR TITLE
add integration tests for pause/unpause upkeeps

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

File: integration-tests/smoke/keeper_test.go [hunk 1]
```diff
@@ -221,6 +227,92 @@ var _ = Describe("Keeper Suite @keeper", func() {
...
+
+			err = chainClient.WaitForEvents()
+			Expect(err).ShouldNot(HaveOccurred(), "Error encountered when waiting for check data update")
+
+			// retrieve old check data for all upkeeps
+			for i := 0; i < len(upkeepIDs); i++ {
+				upkeep, err := registry.GetUpkeepInfo(context.Background(), upkeepIDs[i])
+				Expect(err).ShouldNot(HaveOccurred(), "Failed to get upkeep info at index "+strconv.Itoa(i))
+				Expect(upkeep.CheckData).Should(Equal(newCheckData), "Expect the check data to be %s, but got %s", string(newCheckData), string(upkeep.CheckData))
+			}
```
Comment (infiloop2) [inline integration-tests/smoke/keeper_test.go:247]: This is somewhat tested in contract unit tests. What would be good to test here is that the nodes start using the new checkData to perform upkeeps. 

E.g. you can have an upkeep which returns true only on specific checkData. You can test that the upkeep is not performed initially, but when the checkData changes it starts performing

**Review by infiloop2 — CHANGES_REQUESTED** [review]
Requesting changes to the updateCheckData test

File: integration-tests/smoke/keeper_test.go [hunk 2]
```diff
@@ -221,6 +227,92 @@ var _ = Describe("Keeper Suite @keeper", func() {
...
+				// Obtain the amount of times the upkeep has been executed so far
+				countersAfterPause[i], err = consumers[i].Counter(context.Background())
+				Expect(err).ShouldNot(HaveOccurred(), "Failed to retrieve consumer counter for upkeep at index "+strconv.Itoa(i))
+				log.Info().Msg("Paused upkeep at index " + strconv.Itoa(i) + " which performed " +
+					strconv.Itoa(int(countersAfterPause[i].Int64())) + " times")
+			}
+
+			Consistently(func(g Gomega) {
+				for i := 0; i < len(upkeepIDs); i++ {
+					// Expect the counter to remain constant because the upkeep was paused, so it shouldn't increase anymore
```
[the remark this answers is not in the document]
Comment (FelixFan1992) [reply integration-tests/smoke/keeper_test.go:324]: Done

File: integration-tests/smoke/keeper_test.go (hunk 1 above)
Comment (FelixFan1992) [reply integration-tests/smoke/keeper_test.go]: Done

File: integration-tests/actions/keeper_helpers.go [hunk 3]
```diff
@@ -159,6 +160,61 @@ func DeployPerformanceKeeperContracts(
 	return registry, registrar, upkeeps, upkeepIds
 }
 
+// DeployPerformDataCheckerContracts deploys a set amount of keeper perform data checker contracts registered to a single registry
+func DeployPerformDataCheckerContracts(
```
Comment (infiloop2) [inline integration-tests/actions/keeper_helpers.go:164]: I see there are three functions now all doing the same thing except the keeper contract. Time to refactor and take in upkeep type as an argument?

File: integration-tests/smoke/keeper_test.go [hunk 4]
```diff
@@ -214,13 +223,136 @@ var _ = Describe("Keeper Suite @keeper", func() {
...
 
 		By("Register Keeper Jobs")
 		actions.CreateKeeperJobs(chainlinkNodes, registry)
 		err = chainClient.WaitForEvents()
 		Expect(err).ShouldNot(HaveOccurred(), "Error creating keeper jobs")
 
+		if testToRun == UpdateCheckDataTest {
+			By("tests that counters will be updated after their check data is updated")
+
+			Eventually(func(g Gomega) {
```
Comment (infiloop2) [inline integration-tests/smoke/keeper_test.go:248]: Consistently

**Review by infiloop2 — CHANGES_REQUESTED** [review]
looks good. one small comment on check data test

**Comment by vla33** [comment]
Thanks for adding these tests Felix, will approve as well once that `Eventually` is replaced with `Consistently`.

File: integration-tests/smoke/keeper_test.go (hunk 4 above)
Comment (FelixFan1992) [reply integration-tests/smoke/keeper_test.go]: Done

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**